README.md

```diff
@@ -1,2 +1,185 @@
+    #
+    # Only necessary when not using `.git-branches.toml` to configure perennial branches.
+    # If perennial branches are specified both in `.git-branches.toml` and this input,
+    # the action will prefer `.git-branches.toml` instead.
+    perennial-branches: |
+
+    # Branches matching this regex will be considered
+    # perennial branches.
+    #
+    # Only necessary when not using `.git-branches.toml` to configure perennial branches.
+    # If perennial branches are specified both in `.git-branches.toml` and this input,
+    # the action will prefer `.git-branches.toml` instead.
+    #
+    # Perennial branches specified in `perennial-branches` input will be merged
+    # with branches matching this regex.
+    perennial-regex: ''
+
+    # The template used to generate a visualization of stacked changes for
+    # the current pull request. Template MUST contain `{{ stack }}`, and should
+    # contain valid GitHub-flavoured Markdown.
+    # 
+    # Default: |
+    #   ## Stack
+    #   {{ stack }}
+    template: |
+```
+
+# Guides
+
+## Getting Started
+
+To get started, create a workflow file in `.github/workflows` with the following
+configuration:
+
+```yaml
+name: Git Town
+
+on:
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  git-town:
+    name: Sync Visualization
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      pull-requests: write
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: git-town/action@v1
+```
+
+You can name the workflow file anything you want - we recommend using `git-town.yml`.
+Once configured, the action will visualize your stacked changes whenever a pull request
+is created or updated.
+
+By default, this action will append the visualization to the bottom of the PR body.
+If you want further control over where the visualization appears, simply add start/end
+tags to a [pull request template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository), for example:
+
+```md
+## Description
+
+<!-- stack-start --> 👈
+<!-- stack-end --> 👈
+
+## Checklist
+
+[ ] Foo
+[ ] Bar
+[ ] Baz
+```
+
+This action will use these tags to figure out what sections of the pull request
+body it should replace. The tags will be left behind so that the next time
+this action runs, it will be able to use them again to update the visualization:
+
+```md
+## Description
+
+<!-- stack-start --> 👈 Still there!
+
+## Stack
+
+- `main`
+  - \#1 :point_left:  
+    - \#2
+
+<!-- stack-end --> 👈 Still there!
+
+## Checklist
+
+[ ] Foo
+[ ] Bar
+[ ] Baz
+```
+
+Lastly, since the tags are HTML comments, they will not be visible when viewing 
+the your PR on GitHub.
+
+## Customizing the Visualization
+
+To customize the visualization, simply use the `template` action input:
+
+```yaml
+- uses: git-town/action@v1
+  with:
+    template: |
+      # Stacked Changes
+      {{ stack }}
```

Comment:
I have never used this action, so I might be unaware of usability issues here. You certainly have more experience with this and I trust your judgement. With this said, I wonder whether we really need two separate templating systems here:

1. The `<!-- stack-start -->` and `<!-- stack-end -->` placeholders in the pull request description.
2. The template here that gets inserted between these two tags and uses `{{ stack }}` as a placeholder.

Having two separate placeholders and placeholder formats seems like a lot for such a simple feature. Could we leave templating system 2 out? If the user wants to have a `# Stacked Changes` header above their stacked change in their PR desciption, they can add it to their PR description. Teams can use [pull-request templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository) if they want the pull request descriptions (including the stack visualization in them) to look in a particular way.

Reply:
I can actually answer this question and the previous question around using two anchor tags in one go! The reason is  that if users don't use a PR template, then the visualisation will be added to the PR without a heading! However, thinking through this now, I think this isn't too big of a deal, as the majority of people interested in this action will probably be worked in teams. The chances that teams aren't using a pull request template would be pretty low!

I'll go ahead and make a few changes here:

1. Remove the need to have two anchor tags. We'll use a single tag `<!-- branch-stack -->`, and during updates, we'll look for a list node immediately following the tag and replace everything in between with the updated visualisation.
2. Remove the template input. This will give users the freedom to structure their PR body any way they want. If the tag isn't present, then we'll simply append it to the bottom of the PR body.

Reply:
I like your new approach!